wolfcrypt/test/test.c

```diff
@@ -7255,6 +7265,151 @@ int ecc_encrypt_test(void)
 }
 
 #endif /* HAVE_ECC_ENCRYPT */
+
+#ifdef USE_CERT_BUFFERS_2048
+int ecc_test_buffers() {
+    byte*   tmp;
+    size_t bytes;
+    ecc_key cliKey;
+    ecc_key servKey;
+#ifdef WOLFSSL_CERT_EXT
+    ecc_key keypub;
+#endif
+    WC_RNG rng;
+    word32 idx = 0;
+    int    ret;
+    /* pad our test message to 32 bytes so evenly divisible by AES_BLOCK_SZ */
+    byte   in[] = "Everyone gets Friday off. ecc p";
+    word32 inLen = (word32)XSTRLEN((char*)in);
+    byte   out[256];
+    byte   plain[256];
+    int verify = 0;
+    word32 x;
+#ifdef WOLFSSL_TEST_CERT
+    DecodedCert cert;
+#endif
+
+    tmp = (byte*)TEST_XMALLOC(FOURK_BUF, HEAP_HINT, DYNAMIC_TYPE_TMP_BUFFER);
+    if (tmp == NULL)
+        return -40;
+
+    /* extract the client key from der buffer in wolfssl/certs_test.h */
+    XMEMCPY(tmp, ecc_clikey_der_2048, sizeof_ecc_clikey_der_2048);
+    bytes = sizeof_ecc_clikey_der_2048;
+
+    /* place client key into ecc_key struct cliKey */
+    ret = wc_EccPrivateKeyDecode(tmp, &idx, &cliKey, (word32)bytes);
```

Comment:
You can pass in ecc_clikey_der_2048 and sizeof_ecc_clikey_der_2048 directly. No need to alloc above at 7297.